DataSetExplorer/DataSetBuilder/InstanceFilter.cs

```diff
@@ -0,0 +1,50 @@
+using DataSetExplorer.DataSetBuilder.Model;
+using System.Collections.Generic;
+using CodeModel.CaDETModel.CodeItems;
+using System;
+
+namespace DataSetExplorer.DataSetBuilder
+{
+    internal class InstanceFilter
+    {
+        public List<MetricThresholds> MetricThresholds { get; }
+        
+        public InstanceFilter(List<MetricThresholds> metricThresholds)
+        {
+            MetricThresholds = metricThresholds;
+        }
+
+        public List<Instance> FilterInstances(CodeSmell codeSmell, List<Instance> instances)
+        {
+            var filteredInstances = new List<Instance>();
+            var metricsForSmell = MetricThresholds.FindAll(m => m.CodeSmell.Equals(codeSmell.Name));
+
+            foreach (var i in instances)
+            {
+                if (!ValidSnippetTypeForSmell(i, codeSmell)) continue;
+                var fulfilledConditions = new List<bool>();
+                foreach (var thresholds in metricsForSmell)
+                {
+                    fulfilledConditions.Add(InstanceFulfillesConditions(i, thresholds));
```

Comment:
Can this be simplified to:

    if (!ValidSnippetTypeForSmell(i, codeSmell)) continue;
    if (InstancePassesMetricThresholds(i, metricsForSmell)) filteredInstances.Add(i);


    InstancePassesMetricThresholds can have code similar to "return metricsForSmell.All(metricThreshold => InstanceFulfillesConditions(i, metricThreshold));"